packages/widgets/src/menubar.ts

```diff
@@ -382,6 +391,74 @@ export class MenuBar extends Widget {
     }
   }
 
+  /**
+   * A message handler invoked on an `'resize'` message.
+   */
+  protected _evtResize(event: Event): void {
+    let itemMenus = this.node.getElementsByClassName('lm-MenuBar-item');
+    let screenSize = this.node.offsetWidth;
+    let totalMenuSize = 0;
+    let index = -1;
+    let n = itemMenus.length;
+
+    if (this._menuItemSizes.length == 0) {
+      // Check if it is the first resize and get info about menu items sizes
+      for (let i = 0; i < n; i++) {
+        let item = itemMenus[i] as HTMLLIElement;
+        let found = false;
+        // Add sizes to array
+        totalMenuSize += item.offsetWidth;
+        this._menuItemSizes.push(item.offsetWidth);
+        if (totalMenuSize > screenSize && !found) {
+          index = i;
+          found = true;
+        }
+      }
+    } else {
+      // Calculate current menu size
+      for (let i = 0; i < this._menuItemSizes.length; i++) {
+        totalMenuSize += this._menuItemSizes[i];
+        if (totalMenuSize > screenSize) {
+          index = i;
+          break;
+        }
+      }
+    }
+
+    if (index > -1) {
+      // Create hamburger menu
+      if (this._hamburgerMenu === null) {
+        this._hamburgerMenu = new Menu({ commands: new CommandRegistry() });
+        this._hamburgerMenu.title.label = '...';
+        this._hamburgerMenu.title.mnemonic = 0;
+        this.addMenu(this._hamburgerMenu);
+      }
+
+      // Move menus
+      for (let i = index; i < n - 1; i++) {
+        let submenu = this.menus[i];
+        submenu.title.mnemonic = 0;
+        this._hamburgerMenu.insertItem(0, {
```

Comment:
No, the functionality that I envisioned is that the order of the overflow menu is for the menu items that are left to right to gather top to bottom. Then, it is easier for me to preserve the order when unpacking the elements to the top menu again.

Reply:
Oh I see now why the code doesn't behave the way I thought it would. The resize event fires so many times when resizing the window that it's hard to actually have this loop handle more than one item at a time and when the user shrinks the window, it's like moving backwards through the menu items (because as the window shrinks the menu items get popped off from right to left).